mlflow/system_metrics/metrics/gpu_monitor.py

```diff
@@ -0,0 +1,50 @@
+"""Class for monitoring GPU stats."""
+
+import logging
+import sys
+
+from mlflow.system_metrics.metrics.base_metrics_monitor import BaseMetricsMonitor
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import pynvml
+except ImportError:
+    # If `pynvml` is not installed, a warning will be logged at monitor instantiation.
+    # We don't log a warning here to avoid spamming warning at every import.
+    pass
+
+
+class GPUMonitor(BaseMetricsMonitor):
+    """Class for monitoring GPU stats."""
+
+    def __new__(cls, *args, **kwargs):
+        # Override `__new__` to return a None object if `pynvml` is not installed or GPU is not
+        # found.
+        if "pynvml" not in sys.modules:
+            # Only instantiate if `pynvml` is installed.
+            return
+        try:
+            # `nvmlInit()` will fail if no GPU is found.
+            pynvml.nvmlInit()
+        except pynvml.NVMLError as e:
+            _logger.warning(f"Failed to initialize NVML, skip logging GPU metrics: {e}")
+            return
+        return super().__new__(cls, *args, **kwargs)
```

Comment:
Instead of using `__new__`. Can we add a function like `get_gpu_monitor` that returns None when we fail to import pynvml or `pynvml.nvmlInit()` fails?

Reply:
Having `get_gpu_monitor ` will make `GPUMonitor` diverge from other system metrics modules, I think the current approach is not quite bad? Basically you cannot create a `GPUMonitor`  when missing deps.

Reply:
`GPUMonitor()` returning `None` looks confusing to me. With `get_gpu_monitor`, we don't need to use `__new__`.

Reply:
Agree that returning a `None` from constructor is odd. The bad thing about `get_gpu_monitor` is if we have it, we'd better create `get_cpu_monitor`/`get_disk_monitor` for consistency, which is a bit redundant.

I am swtiching to throw an error, and capture it in `SystemMetricsMonitor` for logging purpose, which should be more clear. Please let me know your thoughts

Reply:
> I am swtiching to throw an error, and capture it in SystemMetricsMonitor for logging purpose, which should be more clear. Please let me know your thoughts

yeah this is clearer.